webapp/plugins/pluggable.jsx

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+// EXPERIMENTAL - SUBJECT TO CHANGE
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class Pluggable extends React.PureComponent {
+    static propTypes = {
+        children: PropTypes.element.isRequired
+    }
+
+    render() {
+        const child = this.props.children.type;
```

Comment:
I like this idea, but is this sort of introspection allowed by RN? I guess worst case we can just structure this like
```
<Pluggable
    name='ProfilePopover'
    default={ProfilePopover}
    {...otherProps}
>
    {children}
</Pluggable>
```